govcd/disk.go

```diff
@@ -339,3 +348,79 @@ func (vdc *Vdc) QueryDisk(diskName string) (DiskRecord, error) {
 
 	return *newDisk, nil
 }
+
+// GetDiskByHref finds a Disk by HREF
+// On success, returns a pointer to the Disk structure and a nil error
+// On failure, returns a nil pointer and an error
+func (vdc *Vdc) GetDiskByHref(diskHref string) (*Disk, error) {
+	util.Logger.Printf("[TRACE] Get Disk By Href: %s\n", diskHref)
+	Disk := NewDisk(vdc.client)
+
+	_, err := vdc.client.ExecuteRequest(diskHref, http.MethodGet,
+		"", "error retrieving Disk: %s", nil, Disk.Disk)
+	if err != nil {
+		return nil, err
+	}
+	return Disk, nil
+}
+
+// GetDiskByName finds a Disk by Name
```

Comment:
This function (and `GetDiskByNameOrId`) may return more than one item. Unlike the other entities we have changed so far, disk allows name duplicates. Only search by ID or HREF guarantee returning a single item.
I suggest either removing the functions that can return duplicates or renaming and modify them to return an array, and then deal with the consequences.

Reply:
From terraform side - we don't allow to create disk with same name, but govcd allows. We have options change `GetName` and `GetDiskByNameOrId` to throw error when two disk with same name exist. Or allow them to return arrays of results.

Reply:
Users can create duplicate disks with other means. Using the GUI (either Flash or HTML5) there is no check for uniqueness.

Reply:
So we can have only these functions and slightly deviate from our *get patter
Get`Disks`ByName which returns aray
Get`Disk`ById
Get`Disk`ByHref

Reply:
Fine with me. @lvirbalas and @Didainius may want to comment.

Reply:
I really like this idea of using plural in a function name which returns an array, it can be generalised as `Get<X>sByName`

Go ahead with the implementation.

@dataclouder please update coding guidelines:
- To reflect this agreement
- To write up `Query...` functions which are to be defined when `/query` API is called

Reply:
Code implemented